src/components/shared/NodeIcon.js

```diff
@@ -6,14 +6,17 @@
 
 import React, { PureComponent } from 'react';
 import explicitConnect from '../../utils/connect';
-import { getIconClassNameForCallNode } from '../../selectors/icons';
+import { getIconClassName } from '../../selectors/icons';
 import { iconStartLoading } from '../../actions/icons';
 
 import type { CallNodeDisplayData } from '../../types/profile-derived';
 import type { ConnectedProps } from '../../utils/connect';
 
 type OwnProps = {|
-  +displayData: CallNodeDisplayData,
+  // This prop is used by call tree.
+  +displayData?: CallNodeDisplayData,
+  // This prop is for other parts of profiler.
+  +iconUrl?: string | null,
 |};
```

Comment:
It would be nice to use the | Flow operator, so that we can type that we should have at least either `displayData` or `iconUrl`, but not both. But I'm not sure how well this works with our tools like eslint and if this works with Flow...

Reply:
Agreed.

Reply:
Thanks, changed it.